vector/src/main/java/im/vector/app/features/home/room/detail/composer/RichTextComposerLayout.kt

```diff
@@ -363,6 +370,9 @@ class RichTextComposerLayout @JvmOverloads constructor(
     }
 
     override fun renderComposerMode(mode: MessageComposerMode) {
+        if (this.composerMode == mode) return
```

Comment:
It's worth noting that `MessageComposerMode.Quote` and `MessageComposerMode.Reply` are not data class, so maybe `==` will not work as expected here.

When replying to someone, then (without sending the message), reply to someone else, does the original sender name updates properly?

Reply:
> It's worth noting that MessageComposerMode.Quote and MessageComposerMode.Reply are not data class, so maybe == will not work as expected here.

I just changed it so they're data classes too (I don't really remember why they weren't in the first place).

> When replying to someone, then (without sending the message), reply to someone else, does the original sender name updates properly?

Yes, both before and after making `Reply` a data class.